thrust/thrust/detail/vector_base.h

```diff
@@ -45,6 +45,11 @@
 
 THRUST_NAMESPACE_BEGIN
 
+struct default_init_t
+{};
+
+constexpr default_init_t default_init;
+
```

Comment:
Given the range of opinions on this feature, I am also fine with wrapping those two entities in an `experimental` namespace.